lib/tool_shed/test/base/twilltestcase.py

```diff
@@ -1253,21 +1059,34 @@ def install_repository(self, name, owner, category_name, install_resolver_depend
         }
         self.visit_url('/repository/install_repositories_by_revision', params=params)
         self.check_for_strings(strings_displayed, strings_not_displayed)
-        # This section is tricky, due to the way twill handles form submission. The tool dependency checkbox needs to
-        # be hacked in through tc.browser, putting the form field in kwd doesn't work.
-        form = tc.browser.get_form('select_tool_panel_section')
+        form = tc.browser.form('select_tool_panel_section')
         if form is None:
-            form = tc.browser.get_form('select_shed_tool_panel_config')
+            form = tc.browser.form('select_shed_tool_panel_config')
         assert form is not None, 'Could not find form select_shed_tool_panel_config or select_tool_panel_section.'
-        kwd = self.set_form_value(form, kwd, 'install_tool_dependencies', install_tool_dependencies)
-        kwd = self.set_form_value(form, kwd, 'install_repository_dependencies', install_repository_dependencies)
-        kwd = self.set_form_value(form, kwd, 'install_resolver_dependencies', install_resolver_dependencies)
-        kwd = self.set_form_value(form, kwd, 'shed_tool_conf', self.shed_tool_conf)
+        kwds = {
+            'install_tool_dependencies': install_tool_dependencies,
+            'install_repository_dependencies': install_repository_dependencies,
+            'install_resolver_dependencies': install_resolver_dependencies,
+            'shed_tool_conf': self.shed_tool_conf,
+        }
         if new_tool_panel_section_label is not None:
-            kwd = self.set_form_value(form, kwd, 'new_tool_panel_section_label', new_tool_panel_section_label)
-        submit_button_control = form.find_control(type='submit')
-        assert submit_button_control is not None, 'No submit button found for form %s.' % form.attrs.get('id')
-        self.submit_form(form.attrs.get('id'), str(submit_button_control.name), **kwd)
+            kwds['new_tool_panel_section_label'] = new_tool_panel_section_label
+        submit_button = [inp.name for inp in form.inputs if getattr(inp, 'type', None) == 'submit']
+        if len(submit_button) == 0:
+            # TODO: refactor, use regular TS install API
+            submit_kwargs = {inp.name: inp.value for inp in tc.browser.forms[0].inputs if getattr(inp, 'type', None) == 'submit'}
+            payload = {_: form.inputs[_].value for _ in form.fields.keys()}
```

Comment:
Very minor, completely subjective, feel free to ignore.  To me, `_` signals that the variable carries no meaning of its own and its sole purpose, if it's a counter variable, is facilitating iteration. But here it is a dictionary key. Maybe give it a generic name, just to emphasize it's not a throwaway? Fine by me either way.

Reply:
I have no preference between `k` and `_`, but I think we use `_ for _ in` more frequently than `k for k in`, so I'd tend go with that form.

Reply:
I am probably responsible for a good chunk of the `_`s in comprehensions, but I now realise that the generally accepted meaning of it is for a throwaway value.
So I will personally not use it in this context for new code.
Thanks for raising it @ic4f !

Reply:
Sounds good. And if at some point we decide to change/prettify any of such code, it would belong in a different PR, of course.